server/src/main/java/org/opensearch/gateway/remote/routingtable/RemoteIndexRoutingTableDiff.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.routingtable;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.store.InputStreamDataInput;
+import org.apache.lucene.store.OutputStreamDataOutput;
+import org.opensearch.cluster.Diff;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.IndexShardRoutingTable;
+import org.opensearch.common.io.Streams;
+import org.opensearch.common.remote.AbstractRemoteWritableBlobEntity;
+import org.opensearch.common.remote.BlobPathParameters;
+import org.opensearch.core.common.io.stream.BufferedChecksumStreamInput;
+import org.opensearch.core.common.io.stream.BufferedChecksumStreamOutput;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.gateway.remote.ClusterMetadataManifest;
+import org.opensearch.index.remote.RemoteStoreUtils;
+import org.opensearch.repositories.blobstore.ChecksumWritableBlobStoreFormat;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.DELIMITER;
+
+/**
+ * Represents a difference between {@link IndexRoutingTable} objects that can be serialized and deserialized.
+ * This class is responsible for writing and reading the differences between IndexRoutingTables to and from an input/output stream.
+ */
+public class RemoteIndexRoutingTableDiff extends AbstractRemoteWritableBlobEntity<RemoteIndexRoutingTableDiff>
+    implements
+        Diff<IndexRoutingTable>,
+        Writeable {
+
+    private final Map<String, Diff<IndexRoutingTable>> diffs;
+
+    private long term;
+    private long version;
+
+    public static final String INDEX_ROUTING_TABLE_DIFF = "index-routing-diff";
+
+    public static final String INDEX_ROUTING_DIFF_METADATA_PREFIX = "indexRoutingDiff--";
+
+    public static final String INDEX_ROUTING_DIFF_FILE = "index_routing_diff";
+    private static final String codec = "RemoteIndexRoutingTableDiff";
+    public static final String INDEX_ROUTING_DIFF_PATH_TOKEN = "index-routing-diff";
+
+    public static final int VERSION = 1;
+
+    public static final ChecksumWritableBlobStoreFormat<RemoteIndexRoutingTableDiff> RemoteIndexRoutingTableDiffFormat =
+        new ChecksumWritableBlobStoreFormat<>(codec, RemoteIndexRoutingTableDiff::new);
```

Comment:
We need to pass read method as second argument, why are we passing constrcutor?

Reply:
We are passing the constructor as the second argument in this case because because it allows the ChecksumWritableBlobStoreFormat to create new instances of RemoteIndexRoutingTableDiff when de-serializing data.

Reply:
Created new non remote entity RoutingTableIncrementalDiff.